src/digest/digest.rs

```diff
@@ -296,6 +306,18 @@ pub struct Algorithm {
     pub nid: c::int,
 }
 
+impl fmt::Debug for Algorithm {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        match self.nid {
+            64 => fmt.write_str("SHA-1"),
+            672 => fmt.write_str("SHA-256"),
+            673 => fmt.write_str("SHA-384"),
+            674 => fmt.write_str("SHA-512"),
+            _ => unreachable!(),
```

Comment:
I just saw the new implementation of this code block after removing the leftover `nid` field at https://github.com/briansmith/ring/commit/eb50203d827c313d650610a34431d90d745009ad#diff-cfa8551ffcaa4c8160640b8a137d65bdR306.

The new implementation is straightforward. Since new `Algorithms` can only be defined in the `ring` crate the implementation behaves correctly.